presto-hive/src/main/java/com/facebook/presto/hive/metastore/SemiTransactionalHiveMetastore.java

```diff
@@ -2074,8 +2074,9 @@ public void run(ExtendedHiveMetastore metastore)
                     Optional<Table> existingTable = metastore.getTable(newTable.getDatabaseName(), newTable.getTableName());
                     if (existingTable.isPresent()) {
                         Table table = existingTable.get();
-                        if (getPrestoQueryId(table).equals(queryId)) {
-                            // ignore table if it was already created by us due to retry
+                        Optional<String> existingTableQueryId = getPrestoQueryId(table);
+                        if (existingTableQueryId.isPresent() && existingTableQueryId.get().equals(queryId)) {
```

Comment:
nit: unsure this is natural, but you can write `existingTableQueryId.equals(Optional.of(queryId))`, avoiding `&&` (and need to extract a variable too)